### DISCUSSION TITLE
Convert to transfer isn't taking the amount of of the source account

### DISCUSSION THREAD

**Discussion by bartekd123** [firefly-iii/firefly-iii#4526] [General]
Hello,

Before I register this as a bug, I wanted to see if this is happening to anyone else that is using the Betas of firefly.

I convert my transactions to a transfer using an automation rule dependant on certain things in the notes/description. Since updating, it looks like it is no longer subtracting the amount of the transaction from the source account. One thing I did notice is when I go in and edit the transaction, when I change the account name, it now includes the current balance of the account, and when I select that, it mentions the note "You're creating a transfer".

Anyone else have this happening? Any workarounds that you can think of to make it work? 

Updated to 5.5.0 beta 3, and still the same issue with a transaction I imported this morning

**Comment by bartekd123** [comment]
Actually I may have figured it out. I had my rule converting it to a transfer, and also setting source/destination. Once I removed the rule setting the source and destination, it seems to have resolved the issue. The rules worked with the non-beta versions of Firefly, so not sure if something was changed.

**Comment by JC5** [comment]
I'll verify later this week. To be clear, your original rule had these actions, in this order?

- Action: Change to transfer with "Account name" as the action value.
- Action: set the source?
- Action: set the destination?

Let me know, so I can replicate it exactly. Or make a screenshot.

**Reply by bartekd123** [reply]
HI James. Thanks for replying. Its not a big deal as it is working for me. However if you want to try to repro and see if it is an issue, this is what I did:

Rule:
 - "When the transaction is created"
 - Trigger - I don't think its important, but it finds it based on description name etc
 - Rule Will: "Convert the transaction to a transfer" & "Set Source account to" -- Both of these rules apply the same account name. I thought they are redundant, so removed the "set source account to" and it works.

I put the rule back, and able to reproduce it, so let me know if you need more details. i tried to look at the transaction details in DEBUG and comparing the 2, but can't see much difference (apart from the obvious timestamps).

**Reply by JC5** [reply]
I can't replicate the exact issue; what I ran into was that the source account became the same as the destination account in case of withdrawals.

Setting the source account will do a better validation from now on so it might just be fixed.
